R/kfold-helpers.R

```diff
@@ -75,27 +71,28 @@ kfold_split_random <- function(K = 10, N = NULL) {
 
 #' @rdname kfold-helpers
 #' @export
-kfold_split_balanced <- function(K = 10, x = NULL) {
+kfold_split_stratified <- function(K = 10, x = NULL) {
   stopifnot(
     !is.null(x),
-    length(unique(x)) == 2,
     K == as.integer(K),
     length(K) == 1,
     K > 1,
     K <= length(x)
   )
-  x <- as.integer(as.factor(x)) - 1
-  x0_id <- which(x == 0)
-  x1_id <- which(x == 1)
-  bins <- rep(NA, length(x))
-  bins[x0_id] <- kfold_split_random(K = K, N = length(x0_id))
-  bins[x1_id] <- kfold_split_random(K = K, N = length(x1_id))
+  x <- as.integer(as.factor(x))
+  Nlev <- length(unique(x))
+  N <- length(x)
+  xids <- numeric()
+  for (l in 1:Nlev) {
+      xids <- c(xids, which(x==l)[sample.int(sum(x==l))])
```

Comment:
I think 
```r 
which(x==l)[sample.int(sum(x==l))]
```
is equivalent to 
```r 
sample(which(x==l), size = 1)
```
right? If so then I think the latter makes the code easier to understand.

Reply:
without size=1 is correct